### PR TITLE
Set up Github Actions

### PR DESCRIPTION
**Pull request by getkey**
Migrate the CI from Google Cloud Build to Github Actions:

1. adds a RBAC so we can make the permissions the CI has access to very tight
2. migrate to Artifacts Registry which allows for scoping projects (and thus tighter permissions)
3. we like Github Actions better in terms of developer experience :slightly_smiling_face: 
4. unify the way code is deployed at @poki

### CODE REVIEW

**Review by koenbollen — COMMENTED** [review]
Hey @getkey! Thanks for this!
I can imagine this is a change to unify the way code is deployed within Poki. And I like this new approach more! 👍
This is however an open-source project, so people will be viewing along (or checking out PRs to determine if they want to use this library).
Could you give a little more context in the PR description? And maybe some more in the deployment files?
Cheers!

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+name: Build
+# more info: https://www.notion.so/poki/How-to-set-up-a-CI-CD-91de5fa03e554aa0bc99fe3b247e8545?pvs=4#bb3a522177744c72adb5dd1608334ede
```
[the remark this answers is not in the document]
Comment (getkey) [reply .github/workflows/build.yaml]: Good point, that's internal documentation on how to deploy services. I don't plan to make it public so I removed it. :slightly_smiling_face:

**Comment by getkey** [comment]
Yes true that, I added a bit more context! :slightly_smiling_face:

**Review by koenbollen — APPROVED** [review]
Cheers!

**Review by erikdubbelboer — APPROVED** [review]

**Pull request merged**